.github/workflows/bloat.yml

```diff
@@ -0,0 +1,137 @@
+name: Cargo Bloat Analysis
+
+on:
+  pull_request:
+  push:
+    branches: [master]
+
+permissions:
+  contents: read
+  statuses: write
+
+jobs:
+  bloat-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: rui314/setup-mold@v1
+
+      - uses: Swatinem/rust-cache@v2
+
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-bloat
+
+      - name: Build base branch
+        id: bloat_base
+        if: github.event_name == 'pull_request'
+        run: |
+          # Build base branch in separate target directory
+          git checkout ${{ github.event.pull_request.base.sha }}
+          CARGO_TARGET_DIR=target-base cargo bloat --release > bloat_base.txt 2>&1 || true
+          git checkout -
+
+          {
+            echo 'output<<EOF'
+            cat bloat_base.txt
+            echo 'EOF'
+          } >> $GITHUB_OUTPUT
+
+      - name: Build PR branch
+        id: bloat_pr
+        run: |
+          # Build PR branch in default target directory
+          cargo bloat --release > bloat_pr.txt 2>&1 || true
+
+          {
+            echo 'output<<EOF'
+            cat bloat_pr.txt
+            echo 'EOF'
+          } >> $GITHUB_OUTPUT
+
+      - name: Compare bloat results
+        id: compare
+        if: github.event_name == 'pull_request'
+        run: |
```

Comment:
What do you think about implementing this shell script as a Python script instead? I think that might be easier to maintain.

Reply:
Sure yeah, a Python script in `.github/scripts/` would work

Reply:
Marked as resolved but left a comment below